### DISCUSSION TITLE
Using Brescianini Controller

### DISCUSSION THREAD

**Discussion by adarsh2798** [bitcraze/discussions#1253] [Q&A]
I wanted to ask how to use ``brescianini controller``. Is there an example anyone has implemeneted?

Seeing the controller file in firmware it need acceleration ,position,velocity(x,y,z) ,and yaw,yawrate as setpoints. Say I edit the code to not use yaw,yawRate,position and velocity setpoints(the desired acceleration is a combination of setpoint->acceleration and velocity and position error, editing out position and velocity setpoints, my desired acceleration is only due to setpoint->acceleration).Now my setpoint structure only needs to have acceleration.x,yz. Is this correct? In such a scenario if I give a command for acceleration, It dosent work. IF crazyflie is hovering at some height and i give a input accelration of (0.1,0,0) it almost falls to ground level and crawls on the ground. This means it is not generating enough thrust. Why is this the case?

**Comment by knmcguire** [comment]
We had to dissect your question a bit as it was unclear but I think we managed.

1- _An example of the Brescianini controller:_ No there is none
2- _Editing the setpoint structure_ I don't think that removing all of those values from the setpoint structure is a good idea as the controller still uses them. If you initialize the setpoint, it will just put those values to zero, and since the controller reacts to that, that will make it drop. The setpoint for position and velocity needs to be the current state of the crazyflie.
3- _The weird behavior when using this setpoint_ See my above answer. Since the controller still uses those setpoints to control the state, the 'actual' position setpoint is now (0,0,0) which explains the behavior. Perhaps send a piece of the code for this?

All in all, I'd like to repeat what I mentioned before in [a discussion](https://github.com/orgs/bitcraze/discussions/1225), we are not giving a guarantee that what you are trying to do (acceleration control without modifying the controllers) will work as this is new territory. It might require a completely new controller.
